source/parser.hera

```diff
@@ -6524,8 +6524,12 @@ Shebang
   /#![^\r\n]*/ EOL
 
 CivetPrologue
-  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter EOS? -> content
-  [\t ]* SingleQuote CivetPrologueContent:content SingleQuote SimpleStatementDelimiter EOS? -> content
+  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter _? EOL? EOS?:eos ->
+    content.children.push(eos)
```

Comment:
Thanks for the PR! One small issue is that the way caching works in the parser we want to avoid mutating the objects passed to the handler rules.

For this case we could do something like:

```javascript
return {...content, children: [...content.children, eos]}
```

Reply:
Thanks! I've incorporated your suggestion verbatim.